pro_tes/ga4gh/tes/task_runs.py

```diff
@@ -410,3 +401,80 @@ def _set_projection(self, view: str) -> Dict:
         else:
             raise BadRequest
         return projection
+
+    def _update_task_incoming(
+        self,
+        payload: dict,
+        db_document: DbDocument,
+        **kwargs
+    ) -> DbDocument:
+        """Update the task incoming object."""
+        logs = self._set_logs(
+            payloads=deepcopy(payload),
+            start_time=kwargs["start_time"]
+        )
+        db_document.task_incoming.logs = [
+            TesTaskLog(**logs) for logs in logs
+        ]
+        db_document.task_incoming.state = TesState.UNKNOWN
```

Comment:
I'm not feeling well and it's not easy to process these changes for me right now. But this seems odd to me. The task state in the incoming task model instance (which we use to return responses to the client) does of course need to be updated from the remote, just like the logs. It shouldn't just remain in state `UNKNOWN`.

But maybe I'm missing something here, it's really hard to concentrate.

Reply:
Well, the state is set to UNKNOWN in the initial setting, while it will be again updated once the task is submitted and it will change from `UNKNOWN` to `QUEUE->RUNNING->COMPLETED` once the background async tracking of the task starts.

Reply:
Okay, fine then :) I'll just merge then, so that you can go on. We can debug/refactor later, if needed.

Reply:
The same goes for the TESexecutor Logs (i.e, logs inside logs ), and logs outputs will update asynchronously, while the other things in logs will remain the same as set in here.